### PR TITLE
[ci] increase Xcode version to enable CircleCI testing for Ruby 3.1

### PR DESCRIPTION
**Pull request by tokizuoh**
### Checklist
- [ ] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [ ] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [ ] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

The CircleCI macOS container holding xcode 13.0.0 does not hold the 3.1 version of Ruby with chruby.[*1](https://circle-macos-docs.s3.amazonaws.com/image-manifest/v6052/index.html) Therefore, Ruby 3.1 has not been tested and is currently being tested with 2.7.4 Ruby.

https://app.circleci.com/pipelines/github/fastlane/fastlane/4899/workflows/634dc941-b190-47dd-af23-e5689286d442/jobs/85785/parallel-runs/0/steps/0-106:
```bash
#!/bin/bash --login -eo pipefail
ruby -v
chruby: unknown Ruby: ruby-3.1  # here
ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-darwin20]
CircleCI received exit code 0
```

The solution was to specify the minimum Xcode version 13.3.1, which allows the Ruby 3.1 version to be selected in chruby, so that the test works correctly.

- Using the macOS execution environment - CircleCI: https://circleci.com/docs/using-macos/
  - Xcode 13.0.0 (current): https://circle-macos-docs.s3.amazonaws.com/image-manifest/v6052/index.html
    - `Installed Rubies 2.6.8, 2.7.4, 3.0.2 (available to chruby)`
  - Xcode 13.3.1: https://circle-macos-docs.s3.amazonaws.com/image-manifest/v7555/index.html
    - `Installed Rubies 2.6.9, 2.7.5, 3.0.3, 3.1.1 (available to chruby)`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -269,8 +269,8 @@ workflows:
           ruby_version: '3.0'
           ruby_opt: -W:deprecated
       - tests_macos:
-          name: 'Execute tests on macOS (Xcode 13.0.0, Ruby 3.1)'
-          xcode_version: '13.0.0'
+          name: 'Execute tests on macOS (Xcode 13.3.1, Ruby 3.1)'
+          xcode_version: '13.3.1'
```
Comment (crazymanish) [inline .circleci/config.yml:273]: why not `Xcode14` ? 🤔

Comment (tokizuoh) [reply .circleci/config.yml:273]: I mistakenly thought I had to test on Xcode13.　Now that I understand that the main purpose is to test on Ruby 3.1, I'll bump it up to Xcode14.

**Comment by tokizuoh** [comment]
# Share current status

## Summary

It's going to take a while to get the test working properly.

## Description
The combination of Ruby 3.1 and Robocop 1.12.1 causes a lot of `Layout/BlockAlignment` errors when running `bundle exec rubocop -a`. This is a known bug and has been released in Robocop 1.22.3 with the following commit.

https://github.com/rubocop/rubocop/commit/1c9abe95132efc43661507521636c75166ea86fa

After updating to Robocop 1.22.3, `bundle install` fails when testing Ruby 2.6 because the dependency `pry-byebug-3.10.1` requires Ruby 2.7 or higher. : [CircleCI log](https://app.circleci.com/pipelines/github/fastlane/fastlane/4924/workflows/13a0ecf8-ff52-4ab8-9c1b-db631f98483a/jobs/85979/parallel-runs/0/steps/0-106)

As a temporary workaround, we have decided to leave the Robocop version unchanged and specify `Layout/BlockAlignment` as `unabled`.　　I haven't run the rspec yet, and will do so in the future.

I will submit a review request when I make progress.
Thanks.

**Comment by ainame** [comment]
👋 Thank you for working on this. I've done a RuboCop update before. We've got `.rubocop_todo.yml` to postpone making fixes for new rules (though it's not ideal). So I'd like to add my two cents on updating RuboCop.

First of all, please try updating RuboCop to the latest. I think it's possible🤔 

> After updating to Robocop 1.22.3, bundle install fails when testing Ruby 2.6 because the dependency pry-byebug-3.10.1 requires Ruby 2.7 or higher

Regarding this problem, I assume you ran `bundle update` without specifying gem names you want to update? If you bump the hardcoded RuboCop version and run the below command, you shouldn't see the update on `pry-byebug` as far as I tested on my end🤔  

```
% bundle update rubocop rubocop-ast rubocop-performance
`` 

And then after updating the RuboCop, please check this page and generate a new `.rubocop_todo.yml` based on the updated version of RuboCop so that you don't need to disable `Layout/BlockAlignment` in `.rubocop.yml`.

https://docs.rubocop.org/rubocop/configuration.html#automatically-generated-configuration

You can use the below options. With `--exclude-limit 0`, you disable all of the rules that that codebase violates in `.rubocop_todo.yml`😇 Not very good, but this at least allows us to keep RuboCop up-to-date without huge effort.
 
```
$ bundle exec rubocop --auto-gen-config --exclude-limit 0
```

See also: https://github.com/fastlane/fastlane/pull/18564

Finally please don't forget to migrate `rubocop.yml` based on warning messages like the below🙏 

> Warning: obsolete parameter `IgnoredMethods` (for `Style/MethodCallWithArgsParentheses`) found in .rubocop.yml
`IgnoredMethods` has been renamed to `AllowedMethods` and/or `AllowedPatterns`.

I hope we get Circle CI jobs right🙏

**Comment by tokizuoh** [comment]
Thanks for telling us about the Rubocop migration😀, @ainame.

> After updating to Robocop 1.22.3, bundle install fails when testing Ruby 2.6 because the dependency pry-byebug-3.10.1 requires Ruby 2.7 or higher : CircleCI log

I was able to solve this problem by the method you showed me and finally bumped to the latest version (1.37.1) of Robocop.

However, as a whole, `bundle exec fastlane execute_tests` causes an error, so I will deal with it 😇

**Comment by tokizuoh** [comment]
以下でrspecの終了ステータスが1になるがエラーの詳細が分からない。（もう少しこのエラーと戦います）

https://github.com/fastlane/fastlane/blob/7de28ca20ebbee386728ea8eb35838a67375a9ca/fastlane/spec/plugins_specs/plugin_generator_spec.rb#L295

**Comment by ainame** [comment]
@tokizuoh I guess it failed because the generated plugin didn't pass RuboCop's checks due to your update.

This line in the template file for a new plugin refers to the RubCop version the fastlane uses. You need to fix the templates to make a plugin generated in RSpec pass RuboCop rules.
https://github.com/fastlane/fastlane/blob/7de28ca20ebbee386728ea8eb35838a67375a9ca/fastlane/lib/fastlane/plugins/template/%25gem_name%25.gemspec.erb#L32

FYI: This is how you create a plugin. You can try it out for testing.
https://docs.fastlane.tools/plugins/create-plugin/

**Comment by tokizuoh** [comment]
@ainame Thanks for the good advice.
The RuboCop test for the generated plugin now passed. All that remains is to get the following runs via Rspec to pass in Ruby 3.1 and all will be green. I'll work on it some more 🏃‍♀️

- `Spaceship::SpaceauthRunner.new.run`
- `Spaceship::SpaceauthRunner.new(username: 'spaceship@krausefx.com').run`

**Comment by tokizuoh** [comment]
It seems that the following return value is false and does not pass rspec, but the cause is not yet known.

https://github.com/fastlane/fastlane/blob/3fa026dcf1fd8ced839fb8703dea2505df6f9106/spaceship/lib/spaceship/client.rb#L414

**Comment by tokizuoh** [comment]
Sorry for leaving it for so long. It was difficult for me. 

I left it in the issue because it will not be resolved as is.
- https://github.com/fastlane/fastlane/issues/20927

**Pull request closed without merging**